scipy/optimize/tests/test__shgo.py

```diff
@@ -56,7 +56,8 @@ class StructTest2(StructTestFunction):
     """
 
     def f(self, x):
-        return (x - 30) * numpy.sin(x)
+        # f must return a scalar
+        return (x[0] - 30) * numpy.sin(x[0])
```

Comment:
There's no point in making these kinds of changes. That is unless there is a test showing that the minimisation goes wrong with before vs after.

Reply:
One cannot optimize functions that return vectors. The change makes the expression mathematically sound and relieves numpy of the burden of having to convert your vector into a scalar in the background (which is what's happening now).